clients/ops/admin-ui/src/features/datastore-connections/add-connection/manual/DSRCustomizationForm.tsx

```diff
@@ -0,0 +1,163 @@
+      validateOnChange={false}
+      validationSchema={Yup.object({
+        fields: Yup.array().of(
+          Yup.object().shape({
+            pii_field: Yup.string().required("PII Field is required"),
+            dsr_package_label: Yup.string().required(
+              "DSR Package Label is required"
+            ),
+          })
+        ),
+      })}
+    >
+      {/* @ts-ignore */}
+      {(props: FormikProps<Values>) => (
+        <Form style={{ marginTop: 0 }} noValidate>
+          <VStack align="stretch">
+            <FieldArray
+              name="fields"
+              render={(fieldArrayProps) => {
+                const { fields } = props.values;
+                return (
+                  <>
+                    <HStack
+                      color="gray.900"
+                      flex="1"
+                      fontSize="14px"
+                      fontWeight="semibold"
+                      lineHeight="20px"
+                      mb="6px"
+                      spacing="24px"
+                    >
+                      <Box w="416px">PII Field</Box>
+                      <Box w="416px">DSR Package Label</Box>
+                      <Box />
+                    </HStack>
+                    <Box
+                      maxH="calc(100vh - 484px)"
+                      paddingRight="13px"
+                      overflow="auto"
+                    >
+                      {fields && fields.length > 0
+                        ? fields.map((_field: Field, index: number) => (
+                            <HStack
+                              // eslint-disable-next-line react/no-array-index-key
+                              key={index}
+                              mt={index > 0 ? "12px" : undefined}
+                              spacing="24px"
+                            >
+                              <Box h="57px" w="416px">
+                                <CustomInput
+                                  autoFocus={index === 0}
+                                  displayHelpIcon={false}
+                                  isRequired
+                                  name={`fields.${index}.pii_field`}
+                                />
+                              </Box>
+                              <Box h="57px" w="416px">
+                                <CustomInput
+                                  displayHelpIcon={false}
+                                  isRequired
+                                  name={`fields.${index}.dsr_package_label`}
+                                />
+                              </Box>
+                              <Box
+                                h="57px"
+                                visibility={index > 0 ? "visible" : "hidden"}
+                              >
+                                <TrashCanSolidIcon
+                                  onClick={() => fieldArrayProps.remove(index)}
+                                  _hover={{ cursor: "pointer" }}
```

Comment:
Actually experimenting with the behavior, I wonder if there should be a trashcan on all blocks, and if they have no fields and click save, we instead delete the underlying connection config instead of creating a webhook with no fields.

Reply:
Interesting! Would we then redirect the user to the Database Connections or Choose your connection screen? Makes sense. Let's talk with @seanpreston about this idea.